### DISCUSSION TITLE
react-native create reuseable myController component

### DISCUSSION THREAD

**Discussion by yossimenash** [react-hook-form/react-hook-form#3235] [General]
Hey, is it possible to create reuseable Controller componet that recives 'name' prop instead of copy-pasting the same Controller tags with diffrent 'name' prop again and again?

i mean instead of writing this:
MyScreen.tsx

```
   <Controller
          name="name"
          control={control}
          rules={{
            required: {value: true, message: 'Name is required'},
          }}
          defaultValue=""
          render={({onChange, value}: any) => (
            <Input
              error={errors.name}
              errorText={errors?.name?.message}
              onChangeText={(text: any) => onChange(text)}
              value={value}
              placeholder="Name"
            />
          )}
        />
       <Controller
          name="email"
          control={control}
          rules={{
            required: {value: true, message: 'Email is required'},
          }}
          defaultValue=""
          render={({onChange, value}: any) => (
            <Input
              error={errors.name}
              errorText={errors?.name?.message}
              onChangeText={(text: any) => onChange(text)}
              value={value}
              placeholder="Email"
            />
          )}
        />

```

write something like this:
MyScreen.tsx
```

<MyControllerComponet name ="name"/>
<MyControllerComponet name ="email"/>
```

and:
MyControllerComponet.tsx.tsx 

```
const MyControllerComponet= (props)=>{
const {handleSubmit, control, errors} = useForm();
 return (
    <View>
      <Controller
        name={props.name}
        control={control}
        rules={{
          required: {value: true, message: 'Email is required'},
          pattern: {
            value: EMAIL_REGEX,
            message: 'Not a valid email',
          },
        }}
        defaultValue=""
        render={({onChange, value}: any) => (
          <TextInput
            style={[styles.inputContainer, props.error, props.style]}
            {...props}
          />
        )}
      />
    </View>
  );
};

```

this example doesn't working - the data of the object doesn't passes from child to parent screen.
how i can make this work?

Thank for helping :)

**Comment by bluebill1049** [comment]
reproduce this problem in a codesandbox plz.

**Reply by yossimenash** [reply]
they dont have react-native option

**Reply by bluebill1049** [reply]
what's wrong with your example? looks fine to me.

**Reply by yossimenash** [reply]
their is a problem with the data transfering.
i want use the data i enter from 'MyControllerComponet.tsx' on 'MyScreen.tsx' screen when i clicking onSubmit. but the data is not forwarded.

**Reply by bluebill1049** [reply]
<img width="903" alt="Screen Shot 2020-10-22 at 8 39 54 pm" src="https://user-images.githubusercontent.com/10513364/96854018-c0667980-14a6-11eb-8b35-3917704be3f5.png">
try to create a CSB with above setting to reproduce the problem

**Reply by yossimenash** [reply]
![stam](https://user-images.githubusercontent.com/44617264/96855016-d81d0e00-1464-11eb-9b83-e97c89800df2.png)

cant see this.

**Reply by yossimenash** [reply]
but here some screenshoots from my project: 
my MainScreen:
![1](https://user-images.githubusercontent.com/44617264/96855536-8923a880-1465-11eb-96b9-d0a35e23a4d6.png)

the exported component which i try to use:
![2](https://user-images.githubusercontent.com/44617264/96855569-92147a00-1465-11eb-88e6-d6f750acdae4.png)

the result:
![3](https://user-images.githubusercontent.com/44617264/96855588-9b054b80-1465-11eb-809a-66ed7c762d05.png)

as you can see, only 1,2 values (the first and second controllers in the MainScreen) logged to console.

**Reply by bluebill1049** [reply]
it's confusing to have the same Controller as name, i would name to something else, also i would try a `naked` Controller first, because add your abstraction. then you would be able to identify the problem.

**Comment by yossimenash** [comment]
if you face the same issue use the following link there is a solution.
https://stackoverflow.com/q/64480040/14491996
